### PR TITLE
Allow baremetal workers with Platform None

### PR DESCRIPTION
**Pull request by sjenning**
This PR allows for the creation of baremetal workers

Remaining issues:
- cluster-api still deploys. `capa-controller-manager` stuck in `ContainerCreating` due to
`MountVolume.SetUp failed for volume "credentials" : secret "node-provider-creds" not found`
- `cluster-autoscaler` runs but continuous error in the logs `E0430 04:42:36.855542       1 reflector.go:138] k8s.io/client-go/dynamic/dynamicinformer/informer.go:91: Failed to watch *unstructured.Unstructured: failed to list *unstructured.Unstructured: machinedeployments.cluster.x-k8s.io is forbidden: User "system:serviceaccount:clusters-example:cluster-autoscaler" cannot list resource "machinedeployments" in API group "cluster.x-k8s.io" at the cluster scope`

example HostedCluster
```
apiVersion: hypershift.openshift.io/v1alpha1
kind: HostedCluster
metadata:
  creationTimestamp: null
  name: example
  namespace: clusters
spec:
  autoscaling: {}
  dns:
    baseDomain: hypershift.example.com
  infraID: example-dev
  initialComputeReplicas: 0
  issuerURL: https://kubernetes.default.svc
  networking:
    machineCIDR: 10.0.0.0/16
    podCIDR: 10.132.0.0/14
    serviceCIDR: 172.31.0.0/16
  platform:
    type: None
  pullSecret:
    name: example-pull-secret
  release:
    image: quay.io/openshift-release-dev/ocp-release:4.8.0-fc.1-x86_64
  services:
  - service: APIServer
    servicePublishingStrategy:
      type: NodePort
      nodePort:
        address: api.example.ocp.example.com
  - service: VPN
    servicePublishingStrategy:
      type: NodePort
      nodePort:
        address: api.example.ocp.example.com
  - service: OAuthServer
    servicePublishingStrategy:
      type: Route
  signingKey:
    name: example-signing-key
  sshKey:
    name: example-ssh-key
```
Create empty NodePool to create MCS
```
$ hypershift create nodepool --node-count=0
```
Resulting NodePool spec
```
apiVersion: v1
kind: NodePool
metadata:
  name: example
  namespace: clusters
spec:
  clusterName: example
  ignitionService:
    type: Route
  nodeCount: 0
  nodePoolManagement:
    autoRepair: false
    maxSurge: 0
    maxUnavailable: 0
  platform:
    type: None
  release:
    image: "" (this is set by the nodepool controller)
```

### CODE REVIEW

**Comment by sjenning** [comment]
@ironcladlou Maybe the diff doesn't do justice to the simplicity. One of the large blocks is just indentation changes.

It is basically just adding `Type` to `NodePoolPlatform` and then only running AWS specific stuff in the nodepool_controller when AWS is the platform.

**Comment by ironcladlou** [comment]
Major question for everyone: how do we automate testing?

File: cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml [hunk 1]
```diff
@@ -197,6 +197,14 @@ spec:
                     required:
                     - instanceType
                     type: object
+                  type:
+                    description: PlatformType is a specific supported infrastructure provider.
+                    enum:
```
Comment (enxebre) [inline cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml:202]: where is this enum coming from? I can't see the validation kubebuilder tag for it in the API field.

File: api/v1alpha1/nodepool_types.go [hunk 2]
```diff
@@ -111,6 +111,7 @@ type NodePoolAutoScaling struct {
 // NodePoolPlatform is the platform-specific configuration for a node
 // pool. Only one of the platforms should be set.
 type NodePoolPlatform struct {
+	Type PlatformType `json:"type"`
```
Comment (enxebre) [inline api/v1alpha1/nodepool_types.go:114]: can we please have kubebuilder enum validation tag here?

File: cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml (hunk 1 above)
Comment (sjenning) [reply cmd/install/assets/hypershift-operator/hypershift.openshift.io_nodepools.yaml:202]: from here https://github.com/openshift/hypershift/blob/d15ea1f33e11c53a1a9124c98e8dadacf333234c/api/v1alpha1/hostedcluster_types.go#L119

**Pull request merged**